code/data_structures/src/tree/heap/min_heap/min_heap.c

```diff
@@ -0,0 +1,110 @@
+#include <stdio.h>
+
+#define asize 100
+
+struct heap
+{
+	int a[asize];
+	int hsize ;
+} H;
+
+void 
+swap(int *a, int *b)
+{
+	int temp = *b;
+	*b = *a;
+	*a = temp;
+}
+
+int 
+min(int a, int b)
+{
+	return (a < b ? a : b);	
+}
+
+void 
+heapInsert(int key)
+{
+	++H.hsize;
+	H.a[H.hsize] = key;
+	int i = H.hsize;
+	while(i > 1 && H.a[i] < H.a[i / 2]) {
+		swap(&H.a[i], &H.a[i / 2]);
+		i = i / 2;
+	}
+	printf("%d is inserted in Min Heap \n", key);
+}
+
+void 
+displayHeap()
+{
+	int i;
+	printf("Elements in the heap:- \n");
+	for(i = 1; i <= H.hsize; i++)
+		printf("%d ", H.a[i]);
+	printf("\n");
+}
+
+void delMin()
+{
+	if(H.hsize == 0) {
+		printf("Heap is empty,No element can be popped out \n");
+		return;
+	}
+
+	int x = H.a[1];
+	H.a[1] = H.a[H.hsize];
+	--H.hsize;
+	int i = 1;
+	
+	while((2 * i) <= H.hsize) {
+		if((2 * i) + 1 <= H.hsize && H.a[i] > min(H.a[2 * i], H.a[(2 * i) + 1])) {
+			if(H.a[(2 * i) + 1] < H.a[2 * i]) {
+				swap(&H.a[i], &H.a[(2 * i) + 1]);
```

Comment:
``` swap(&H.a[i], &H.a[(2 * i) + 1]); ```

You could write this as macro as well.

    #define swap (array,first,second) array[second] = array[first] + 0 * ( array[first] = array[second] )
    swap(H.a , i , 2*i+1);